jenkins/bootstrap.py

```diff
@@ -330,10 +330,18 @@ def cat(self, path, generation):
         return self.call(cmd, output=True)
 
 
-    def upload_artifacts(self, path, artifacts):
+    def upload_artifacts(self, gsutil, path, artifacts):
         """Upload artifacts to the specified path."""
         # Upload artifacts
         if os.path.isdir(artifacts):
+            if os.path.basename(path) == os.path.basename(artifacts):
+                try:
+                    gsutil.stat(path)
+                    # Success means remote path exists
+                    base = os.path.basename(path) + '/'
+                    path = path[:-len(base)]
```

Comment:
it should be https://github.com/kubernetes/test-infra/blob/57a5fc6e02067974a382b63ea863ddf021d0eee8/jenkins/bootstrap.py#L551